components/ai-services-mgt/org.wso2.carbon.ai.service.mgt/src/main/java/org/wso2/carbon/ai/service/mgt/token/AIAccessTokenManager.java

```diff
@@ -0,0 +1,265 @@
+ */
+public class AIAccessTokenManager {
+
+    private static volatile AIAccessTokenManager instance;  // Volatile for thread safety.
+    private static final Object lock = new Object();  // Lock for synchronization.
+
+    private static final Log LOG = LogFactory.getLog(AIAccessTokenManager.class);
+
+    private static final String AI_KEY = IdentityUtil.getProperty(AI_SERVICE_KEY_PROPERTY_NAME);
+    private static final String AI_TOKEN_ENDPOINT = IdentityUtil.getProperty(AI_TOKEN_ENDPOINT_PROPERTY_NAME);
+
+    private AccessTokenRequestHelper accessTokenRequestHelper;
+
+    private String accessToken;
+    private final String clientId;
+
+    private AIAccessTokenManager() {
+
+        byte[] decodedBytes = Base64.getDecoder().decode(AI_KEY);
+        String decodedString = new String(decodedBytes, StandardCharsets.UTF_8);
+        String[] parts = decodedString.split(":");
+        if (parts.length == 2) {
+            this.clientId = parts[0]; // Extract clientId.
+        } else {
+            throw new IllegalArgumentException("Invalid AI service key.");
+        }
+    }
+
+    /**
+     * Get the singleton instance of the AIAccessTokenManager.
+     *
+     * @return The singleton instance.
+     */
+    public static AIAccessTokenManager getInstance() {
+
+        if (instance == null) {
+            synchronized (lock) {
+                if (instance == null) {
+                    instance = new AIAccessTokenManager();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Set the access token request helper.
+     *
+     * @param helper The access token request helper.
+     */
+    protected void setAccessTokenRequestHelper(AccessTokenRequestHelper helper) {
+
+        this.accessTokenRequestHelper = helper;
+    }
+
+    /**
+     * Get the access token.
+     *
+     * @param renewAccessToken Whether to renew the access token.
+     * @return The access token.
+     * @throws AIServerException If an error occurs while obtaining the access token.
+     */
+    public String getAccessToken(boolean renewAccessToken) throws AIServerException {
+
+        if (StringUtils.isEmpty(accessToken) || renewAccessToken) {
+            synchronized (AIAccessTokenManager.class) {
+                if (StringUtils.isEmpty(accessToken) || renewAccessToken) {
+                    this.accessToken = accessTokenRequestHelper != null ?
+                            accessTokenRequestHelper.requestAccessToken() : createDefaultHelper().requestAccessToken();
+                }
+            }
+        }
+        return this.accessToken;
+    }
+
+    private AccessTokenRequestHelper createDefaultHelper() {
+
+        return new AccessTokenRequestHelper(AI_KEY, AI_TOKEN_ENDPOINT,
+                // Here we keep the default HTTP client to send the token request.
+                // We open and close it for each request.
+                HttpAsyncClients.createDefault());
```

Comment:
I don't see a reason to use the Async Client here. Lets check and switch to closable http client